src/modules/NtpClient.rb

```diff
@@ -1045,42 +1045,53 @@ def Summary
       summary
     end
 
+
     # Test if specified NTP server answers
     # @param [String] server string host name or IP address of the NTP server
+    # @return [Boolean] true if NTP server answers properly
+    def reachable_ntp_server?(server)
+      # testing the server using IPv4 and then using IPv6 protocol
+      # bug #74076, Firewall could have been blocked IPv6
+      return true if sntp_test(server) == 0
+
+      sntp_test(server, 6) == 0
+    end
+
+    # -K /dev/null: use /dev/null as KoD history file (if not specified,
+    #               /var/db/ntp-kod will be used and it doesn't exist)
+    # -c: concurrently query all IPs; -t 5: five seconds of timeout
+    # @param [String] server string host name or IP address of the NTP server
+    # @param [Fixnum] integer ip version to use (4 or 6)
+    # @return [Fixnum] sntp exit code or 1 in case of lookup error
+    def sntp_test(server, ip_version = 4)
```

Comment:
Thinking twice about this method. It's strange to have a method that returns a possibly modified exit code. At the end of the day, we are always comparing it's output with 0, so I would make it boolean.

A method that returns "low level stuff" like exit codes should not mess with that info. It's unexpected.